lib/gollum-lib/wiki.rb

```diff
@@ -207,13 +195,7 @@ def file(name, version = @ref, try_on_disk = false)
     #
     # Returns the in-memory Gollum::Page.
     def preview_page(name, data, format)
-      page = ::Gollum::Page.new(self)
-      ext  = ::Gollum::Page.format_to_ext(format.to_sym)
-      filename = "#{name}.#{ext}"
-      blob = OpenStruct.new(:name => filename, :data => data, :is_symlink => false)
-      page.populate(blob)
-      page.version = @access.commit(@ref)
-      page
+      ::Gollum::PreviewPage.new(self, "#{name}.#{::Gollum::Page.format_to_ext(format.to_sym)}", data, @access.commit(@ref))
```

Comment:
Clean. The only worry I have is that the method signature for initialization is not completely intuitive, but I do like how short it is.